contexts/NotificationsContext.tsx

```diff
@@ -0,0 +1,88 @@
+import { createContext, ReactNode, useState } from "react";
+import { AddNotificationFn, NotificationT, RemoveNotificationFn } from "types";
+
+export interface NotificationsContextState {
+  notifications: NotificationT[];
+  addNotification: (notification: NotificationT) => void;
+  addSuccessNotification: AddNotificationFn;
+  addErrorNotification: AddNotificationFn;
+  addPendingNotification: AddNotificationFn;
+  removeNotification: RemoveNotificationFn;
+  clearNotifications: () => void;
+}
+
+export const defaultNotificationsContextState: NotificationsContextState = {
+  notifications: [],
+  addNotification: () => null,
+  addSuccessNotification: () => null,
+  addErrorNotification: () => null,
+  addPendingNotification: () => null,
+  removeNotification: () => null,
+  clearNotifications: () => null,
+};
+
+export const NotificationsContext = createContext(defaultNotificationsContextState);
+
+export function NotificationsProvider({ children }: { children: ReactNode }) {
+  const [notifications, setNotifications] = useState<NotificationT[]>([]);
+
+  function addNotification(notification: NotificationT) {
+    setNotifications((prev) => [...prev, notification]);
+  }
+
+  function clearNotifications() {
+    setNotifications([]);
+  }
+
+  function removeNotification(transactionHash: string) {
+    setNotifications((prev) => prev.filter((prevNotification) => prevNotification.transactionHash !== transactionHash));
+  }
+
+  function addSuccessNotification(description: ReactNode, transactionHash: string) {
+    addNotification({
+      description,
+      transactionHash,
+      type: "success",
+    });
+
+    setTimeout(() => {
+      removeNotification(transactionHash);
+    }, 5000);
```

Comment:
Autohide settled notifications. We don't need to auto-hide pending ones because they are removed once the mutation is settled.